simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/HeartbeatScheduler.java

```diff
@@ -19,22 +21,39 @@
     private StationState stationState;
     private StationMessageSender stationMessageSender;
 
+    private AtomicInteger heartBeatInterval;
+    private AtomicLong timeOfLastHeartbeatSent;
+
     public HeartbeatScheduler(StationState stationState, StationMessageSender stationMessageSender) {
         this.stationState = stationState;
         this.stationMessageSender = stationMessageSender;
+        this.heartBeatInterval = new AtomicInteger();
+        this.timeOfLastHeartbeatSent = new AtomicLong(0);
     }
 
-    public void scheduleHeartbeat(int heartbeatInterval) {
+    public void updateHeartbeat(int heartbeatInterval) {
         LOGGER.debug("Scheduling heartbeat to {} sec.", heartbeatInterval);
-        heartbeatExecutor.scheduleAtFixedRate(this::sendHeartbeat, heartbeatInterval, heartbeatInterval, TimeUnit.SECONDS);
+        heartBeatInterval.set(heartbeatInterval);
+        heartbeatExecutor.scheduleAtFixedRate(this::shouldSendHeartbeat, heartbeatInterval, heartbeatInterval, TimeUnit.SECONDS);
```

Comment:
Consider passing heartbetInterval to the method, no need to declare it as a field